File: scaffold/src/main/java/com/yullg/android/scaffold/ui/canvas/ReverseMode.java
package com.yullg.android.scaffold.ui.canvas;

import androidx.annotation.IntDef;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * 定义翻转方式
 */
@Documented
@Retention(RetentionPolicy.CLASS)
@IntDef(value = {ReverseMode.NONE, ReverseMode.HORIZONTAL, ReverseMode.VERTICAL}, flag = true)
public @interface ReverseMode {
    int NONE = 0;
    int HORIZONTAL = 1;
    int VERTICAL = 2;
}

File: scaffold/src/main/java/com/yullg/android/scaffold/ui/chart/ReverseMode.kt
package com.yullg.android.scaffold.ui.chart

enum class ReverseMode {
    NONE, HORIZONTAL, VERTICAL
}

fun Int.toReverseMode() = let {
    when (it) {
        1 -> ReverseMode.HORIZONTAL
        2 -> ReverseMode.VERTICAL
        else -> ReverseMode.NONE
    }
}